src/Storages/System/StorageSystemJemalloc.cpp

```diff
@@ -25,6 +25,12 @@ UInt64 getJeMallocValue(const char * name)
     UInt64 value{};
     size_t size = sizeof(value);
     mallctl(name, &value, &size, nullptr, 0);
+#if __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+    if (size == 4)
+    {
+        value >>= 32;
```

Comment:
Why don't fix jemalloc instead?

Reply:
@alexey-milovidov, jemalloc does the correct thing: for some queries like "arenas.nbins", mallctl() fills the output with 4 bytes of integer value and ALSO set the "size" to 4.
The problem is that the code was using an UInt64 variable for an 32 bit output. On little-endian machines it is correct, but on big-endian machine the value would be 2^32 times larger than the actual result.
The fix checks if the actual output size (which is set by mallctl()) is 4. If it is, we need to right-shift the UInt64 integer by 32 on big-endian machines. (Another way to fix it is to use UInt32 for output for those 32 bit values according to names, but the current fix is much more elegant).